modules/gateway/peers.go

```diff
@@ -126,6 +137,16 @@ func (g *Gateway) threadedAcceptConn(conn net.Conn) {
 		return
 	}
 
+	if build.VersionCmp(remoteVersion, sessionHandshakeUpgradeVersion) >= 0 {
+		wantConnect := true
+		err = acceptConnSessionHandshake(conn, g.id, wantConnect)
```

Comment:
I think `wantConn` is implied to always be true for `acceptConnSessionHandshake`, so it doesn't need to take it as an argument. If we want to explicitly reject a peer we could use a new function called `reject` instead of `accept`.